libsolidity/analysis/TypeChecker.cpp

```diff
@@ -291,42 +291,70 @@ void TypeChecker::checkContractAbstractFunctions(ContractDefinition const& _cont
 			}
 }
 
-void TypeChecker::checkContractAbstractConstructors(ContractDefinition const& _contract)
+void TypeChecker::annotateBaseConstructorArguments(
+	ContractDefinition const& _contract,
+	FunctionDefinition const* _baseConstructor,
+	ASTNode const* _node
+)
 {
-	set<ContractDefinition const*> argumentsNeeded;
-	// check that we get arguments for all base constructors that need it.
-	// If not mark the contract as abstract (not fully implemented)
+	bool const v050 = _contract.sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050);
+
+	auto insertionResult = _contract.annotation().baseConstructorArguments.insert(std::make_pair(_baseConstructor, _node));
+	if (!insertionResult.second)
+	{
+		ASTNode const* previousNode = insertionResult.first->second;
+
+		SecondarySourceLocation ssl;
+		ssl.append("Second constructor call is here:", _node->location());
+
+		if (v050)
+			m_errorReporter.declarationError(
+				previousNode->location(),
+				ssl,
+				"Duplicated super constructor call."
```

Comment:
Actually it is not really a call (we cannot change the order of constructor executions, for example). Perhaps `Base constructor arguments given twice`?